src/PotentialSimulation/Painting/Painting.jl

```diff
@@ -210,4 +210,14 @@ function paint!(point_types, potential, face::AbstractSurfacePrimitive{T}, geome
         end
     end 
     nothing
+end
+
+function mark_surface_bits!(point_types::Array{PointType, 3})
+    i1max, i2max, i3max = size(point_types)
+    for i in findall(point_types .& pn_junction_bit .> 0)
+        i1, i2, i3 = i[1], i[2], i[3]
+        point_types[i1,i2,i3] += surface_bit * !(all([point_types[j1,j2,j3] & pn_junction_bit > 0 
```

Comment:
Here, the `surface_bit` is set for all points with `pn_junction_bit` if not all neighboring points are inside the pn-junction. Changing `+=` to `=` in line 219, we could set the `pn_junction_bit` to 0.